tot/data_processing/scaler.py

```diff
@@ -0,0 +1,219 @@
+        """
+        IDs = df["ID"].unique()
+        df_pivot = _pivot(df, col_name)
+
+        df_pivot[IDs] = self.transformer.inverse_transform(df_pivot[IDs])
+
+        return _melt(df_pivot, IDs, col_name)
+
+    def _rescale(self, df: pd.DataFrame, col_name: str) -> pd.DataFrame:
+        """
+        Applies `inverse_transform` on column `col_name` in `df`.
+
+        Parameters:
+        -----------
+        df : pd.DataFrame
+            dataframe containing column ``ds``, ``y``, [``yhat<i>``], and optionally ``ID`` with data
+        col_name : str
+            name of the column, on which the operation is applied
+
+        Returns:
+        --------
+        pd.DataFrame
+            dataframe containing column ``ds``, ``y``, [``yhat<i>``],  and optionally ``ID`` with rescaled data
+        """
+        df[col_name] = self.transformer.inverse_transform(df[col_name].values.reshape(-1, 1))
+        return df
+
+    def _inverse_transform(self, df: pd.DataFrame) -> pd.DataFrame:
+        """
+        Applies rescaling on the `df`. First, rescaling is performed on the `y` column to create the main df. Then,
+        operation is repeated on all `yhat` values and results are updated in the main df. Proper `rescale`
+        implementation is chosen based on scaling level.
+
+        Parameters:
+        -----------
+        df : pd.DataFrame
+            dataframe containing column ``ds``, ``y``, [``yhat<i>``] and optionally ``ID`` with data
+
+        Returns:
+        --------
+        pd.DataFrame
+            dataframe containing column ``ds``, ``y``, [``yhat<i>``] and optionally ``ID`` with rescaled data
+        """
+        if self.scaling_level == "per_time_series":
+            rescale_method = self._rescale_per_series
+        else:
+            rescale_method = self._rescale
+
+        result = rescale_method(df, "y")
+
+        yhats = [col for col in df.columns if "yhat" in col]
+        for yhat in yhats:
+            result[yhat] = rescale_method(df, yhat)[yhat]
+
+        return result
+
+    def transform(self, df_train: pd.DataFrame, df_test: pd.DataFrame) -> Tuple[pd.DataFrame, pd.DataFrame]:
+        """
+        Applies `transform` on the dataframes. The transformer is fit on the `df_train`.
+
+        Parameters:
+        -----------
+        df_train : pd.DataFrame
+            dataframe containing column ``ds``, ``y``, and optionally ``ID`` with train data
+        df_train : pd.DataFrame
+            dataframe containing column ``ds``, ``y``, and optionally ``ID`` with test data
+
+        Returns:
+        --------
+        pd.DataFrame
+            dataframe containing column ``ds``, ``y``, and optionally ``ID`` with scaled train data
+        pd.DataFrame
+            dataframe containing column ``ds``, ``y``, and optionally ``ID`` with scaled test data
+        """
+        df_train = df_train.copy()
+        df_test = df_test.copy()
+        if self.scaling_level == "per_time_series":
```

Comment:
@ankke A minor detail. Can we streamline the query for `if self.scaling_level == "per_time_series"`  for the `transfom()` and `inverse_transform()`? It is queried in the `transform()`, however for the inverse it is queried in the `_inverse_transform()`